src/registry/currency/edit.tsx

```diff
@@ -0,0 +1,214 @@
+      </TabPanel>
+
+      {/* Advanced tab */}
+      <TabPanel>
+        <SimpleConditional />
+      </TabPanel>
+
+      {/* Validation tab */}
+      <TabPanel>
+        <Validate.Required />
+        <Validate.ValidatorPluginSelect />
+        <Validate.Min />
+        <Validate.Max />
+        <Validate.ValidationErrorTranslations />
+      </TabPanel>
+
+      {/* Registration tab */}
+      <TabPanel>
+        <Registration.RegistrationAttributeSelect />
+      </TabPanel>
+
+      {/* Translations */}
+      <TabPanel>
+        <Translations.ComponentTranslations<CurrencyComponentSchema>
+          propertyLabels={{
+            label: intl.formatMessage(LABELS.label),
+            description: intl.formatMessage(LABELS.description),
+            tooltip: intl.formatMessage(LABELS.tooltip),
+          }}
+        />
+      </TabPanel>
+    </Tabs>
+  );
+};
+
+EditForm.defaultValues = {
+  // basic tab
+  label: '',
+  key: '',
+  description: '',
+  tooltip: '',
+  showInSummary: true,
+  showInEmail: false,
+  showInPDF: true,
+  hidden: false,
+  clearOnHide: true,
+  isSensitiveData: false,
+  defaultValue: undefined,
+  decimalLimit: undefined,
+  allowNegative: false,
+  currency: 'EUR',
+  // Advanced tab
+  conditional: {
+    show: undefined,
+    when: '',
+    eq: '',
+  },
+  // Validation tab
+  validate: {
+    required: false,
+    plugins: [],
+    min: undefined,
+    max: undefined,
+  },
+  translatedErrors: {},
+  // Registration tab
+  registration: {
+    attribute: '',
+  },
+};
+
+const DefaultValue: React.FC = () => {
+  const intl = useIntl();
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'defaultValue' builder field",
+    defaultMessage: 'This will be the initial value for this field before user interaction.',
+  });
+  return (
+    <NumberField
+      name="defaultValue"
+      label={<FormattedMessage {...LABELS.defaultValue} />}
+      tooltip={tooltip}
+    />
+  );
+};
+
+const DecimalLimit: React.FC = () => {
+  const intl = useIntl();
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'decimalLimit' builder field",
+    defaultMessage: 'The maximum number of decimal places.',
+  });
+  return (
+    <NumberField
+      name="decimalLimit"
+      label={
+        <FormattedMessage
+          description="Label for 'decimalLimit' builder field"
+          defaultMessage="Decimal places"
+        />
+      }
+      tooltip={tooltip}
+    />
+  );
+};
+
+const AllowNegative: React.FC = () => {
+  const intl = useIntl();
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'allowNegative' builder field",
+    defaultMessage: 'Allow negative values.',
+  });
+  return (
+    <Checkbox
+      name="allowNegative"
+      label={
+        <FormattedMessage
+          description="Label for 'allowNegative' builder field"
+          defaultMessage="Allow negative values"
+        />
+      }
+      tooltip={tooltip}
+    />
+  );
+};
```

Comment:
Should we move this to `components/builder`?

Reply:
let's not yet - they are not generic, just happen to be used in both currency/number component but those are basically parent & child. AllowNegative is scheduled for removal too.